lib/plugins/logviewer/admin.php

```diff
@@ -90,36 +90,39 @@ protected function displayLog()
             echo $this->locale_xhtml('nolog');
             return;
         }
+        
+        $logfileSize = filesize($logfile) / 1_048_576; // in MB
+        $lines = [];
+
+        if($logfileSize > 128) { // PHP default memory limit            
+            $lines = $this->readEndOfLogFile($logfile, 1500);
+            
+            // remove incomplete lines from the beginning
+            foreach($lines as $line) {
+                if (substr($line, 0, 2) === '  ') {
+                    array_shift($lines);
+                } else {
+                    break;
+                }
+            }
 
-        // loop through the file an print it
-        echo '<dl>';
-        $lines = file($logfile);
-        $cnt = count($lines);
-        for ($i = 0; $i < $cnt; $i++) {
-            $line = $lines[$i];
-
-            if (substr($line, 0, 2) === '  ') {
-                // lines indented by two spaces are details, aggregate them
-                echo '<dd>';
-                while (substr($line, 0, 2) === '  ') {
-                    echo hsc(substr($line, 2)) . '<br />';
-                    $i++;
-                    $line = $lines[$i] ?? '';
+            // create the url to log file
+            $pathItems = explode('/', $logfile);
+            foreach($pathItems as $item) {
+                if($item != 'data') {
+                    array_shift($pathItems);
+                } else {
+                    break;
                 }
-                echo '</dd>';
-                $i -= 1; // rewind the counter
-            } else {
-                // other lines are actual log lines in three parts
-                list($dt, $file, $msg) = sexplode("\t", $line, 3, '');
-                echo '<dt>';
-                echo '<span class="datetime">' . hsc($dt) . '</span>';
-                echo '<span class="log">';
-                echo '<span class="msg">' . hsc($msg) . '</span>';
-                echo '<span class="file">' . hsc($file) . '</span>';
-                echo '</span>';
-                echo '</dt>';
             }
+            $logURL = '/' . implode('/', $pathItems);
+            echo "<p><span style='color:red;'>WARNING</span>: the file is too large, <a href='{$logURL}' target='_blank'>click here</a> to see the complete version.</p>";
```

Comment:
The data dir is not directly accessible. You can not simply link to the log file. It needs a proper data dispatcher that also checks admin permissions (log files should only be available to admins).

Also please do not use inline styles. Simplest is probably using DokuWiki's `msg()` method here.